.github/workflows/ubi8-nightly.yml

```diff
@@ -0,0 +1,40 @@
+name: github-UBI8-NIGHTLY
+
+# Runs every night at midnight
+on:
+  schedule:
+    - cron: '00 00 * * *'
+
+permissions:
+  contents: none
+
+# Cancels any in progress 'workflow' associated with this PR
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  ubi8-nightly:
+    name: ubi8-nightly
+    runs-on: [ubuntu-latest]
+    permissions:
+      packages: read
+      contents: read
+    container: 
+      image: ghcr.io/sandialabs/opencsp:latest-ubi8
+      credentials:
+        username: ${{ github.actor }}
+        password: ${{ secrets.GITHUB_TOKEN }}
+
+    steps:
+      - name: checkout
+        uses: actions/checkout@v4
+        with:
+          path: OpenCSP
+
+      - name: pytest-cov
+        working-directory: OpenCSP/example
+        run: |
+          python3 -m pip install -r ../requirements.txt
```

Comment:
This reasoning makes sense to me. I don't know the implications to runtime/using resources of this though. Those would be my only concerns but I'm not well versed in this.

Reply:
No, this runs pip install in the container image.

A image would be built with something like: `docker build ...` or `podman build ...`. For example, see https://github.com/sandialabs/OpenCSP/blob/main/.github/workflows/docker.yml.

Reply:
@bbean23: In general, I think this is a good idea. That being said, I think updating everything in the image every night is asking for trouble. If you would like to triage the nightlies when these fail, then I am OK with it. Otherwise, I think it would be best to setup a second nightly that updates everything in a separate image and runs examples + unit tests. Even with that setup, I am not volunteering for keeping this running : )

Reply:
Yep, a second nightly is exactly what I'm proposing. And I agree, you shouldn't be necessarily responsible for making it work. In fact, it would be helpful for me to learn how to set up a container like this. I'll make a ticket to do so.